public/script.js

```diff
@@ -17,3 +17,72 @@ if ("serviceWorker" in navigator) {
       console.error(err);
     });
 }
+
+function userAgentDetectDevice(){
+  // if the userAgent contains "Windows"
+  if(navigator.userAgent.indexOf("Windows") !== -1){
+    return "Windows"
+  }
+  // if the userAgent contains "Macintosh"
+  else if(navigator.userAgent.indexOf("Macintosh") !== -1){
+    return "macOS"
+  }
+  // if the userAgent contains "Linux"
+  else if(navigator.userAgent.indexOf("Linux") !== -1){
+    return "Linux"
+  }
+  // if the userAgent contains "Android"
+  else if(navigator.userAgent.indexOf("Android") !== -1){
+    return "Android"
+  }
+  // if the userAgent contains "iPhone" or "iPad" or "iPod"
+  else if(navigator.userAgent.indexOf("iPhone") !== -1 || navigator.userAgent.indexOf("iPad") !== -1 || navigator.userAgent.indexOf("iPod") !== -1){
+    return "iOS"
+  }
+}
+
+function detectDevice(){
+  // try to detect using navigator.userAgentData.platform
+  try{
+    // windows
+    if(navigator.userAgentData.platform === "Windows"){
```

Comment:
I won't block the review based on this comment.

Is it worth turning this into a `switch/case`, since we're just checking the same property each time?